multiversx/mx-chain-go
# added map support for over writable configs

## Reasoning behind the pull request
- Needed support for Map in over writable configs for `enableRound.toml`
- also cleaned some stuff in tests
  
## Proposed changes
- Added support for `Map` to be overwritten by other map
- Added tests with real data from a toml file

## Testing procedure
- uncomment and always use `OverridableConfigTomlValues` in `cmd/node/config/prefs.toml` 
- test override map with real data from prefs.toml (similar examples in `testscommon/toml/overwrite.toml`)
- also test other over-writable values: int/float/bool/struct/slice etc - with good and bad data
- nodes should start normally with correct values and log what was overridden, otherwise not start

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?
